Fix remind module to allow custom prefixes

demophoon opened the pull request:
Before this commit the remind module only worked if you had the default
prefix set. This commit created the remind regex during the setup step
so that we can read jenni.config.prefix and initialize r_command with
the proper prefix.

kaneda commented on modules/remind.py:
```diff
@@ -109,10 +79,48 @@ def monitor(jenni):
+    periods = '|'.join(scaling.keys())
+    p_command = r'{}in ([0-9]+(?:\.[0-9]+)?)\s?((?:{})\b)?:?\s?(.*)'.format(
+        jenni.config.prefix,
+        periods,
```
What's with the hanging comma?

demophoon replied:
I do that so that if in the future if these an item is added or removed from this function call git diff will show only the lines added or removed instead of n+1 lines for fixing the additional comma on the ending line.

kaneda commented on modules/remind.py:
```diff
@@ -109,10 +79,48 @@ def monitor(jenni):
+    # jenni.sending.release()
+
+    def monitor(jenni):
+        time.sleep(5)
```
Yuck

kaneda commented on modules/remind.py:
```diff
@@ -109,10 +79,48 @@ def monitor(jenni):
+                # jenni.sending.release()
+            time.sleep(2.5)
+
+    targs = (jenni,)
```
Likewise, floating coma

demophoon replied:
That is how singleton tuples are suppose to be constructed, see https://docs.python.org/2/tutorial/datastructures.html#tuples-and-sequences.

`threading.Thread`'s args argument accepts tuples to pass into its `__init__` function which is then unpacked and sent into the `monitor` function as arguments.

kaneda commented:
@myano what do you think?

demophoon commented:
@kaneda I've gone back and split out that one commit so that the commits are more representative of what I actually changed. Before I moved a large dictionary around the setup function to make it seem more logical the setup function will have all of the variables it needs defined before the setup function itself is defined.

For some reason git thought I rewrote the entire function instead of just moving code around. The new commits are below.

kaneda commented:
@demophoon I think this looks good, but I'm going to wait for @myano to approve

demophoon commented:
Is there anything else that needs to be done here @kaneda or @myano?

kaneda commented:
@demophoon still waiting for @myano to weigh in, he should be around tomorrow night
